exec/java-exec/src/main/java/org/apache/drill/exec/coord/local/LocalClusterCoordinator.java

```diff
@@ -125,6 +127,11 @@ public DistributedSemaphore getSemaphore(final String name, final int maximumLea
     return semaphores.get(name);
   }
 
+  @Override
+  public <V> TransientStore<V> getOrCreateTransientStore(final TransientStoreConfig<V> config) {
+    return new MapBackedStore<>(config);
```

Comment:
Looks like we are creating a new MapBackedStore for every call to this. There is no caching factory in the call path to this method either.